ProcessLib/LIE/Common/BranchProperty.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <Eigen/Eigen>
+
+namespace ProcessLib
+{
+namespace LIE
+{
+struct BranchProperty
+{
+	int node_id;
+	Eigen::Vector3d coords;
+	int master_fracture_ID;
+	int slave_fracture_ID;
+	// unit vector normal to the master fracture in a direction to the slave
+	Eigen::Vector3d normal_vector_branch;
+
+    virtual ~BranchProperty() = default;
```

Comment:
- Since nothing is derived from this class, I'd drop the dtor and mark the class `final`.
 - Reorder the members as {coords, normal, node_id, further ints} s.t. to reduce the holes in the memory because of alignment.
 - Not sure, but might be necessary to have aligned new operator http://eigen.tuxfamily.org/dox/group__TopicStructHavingEigenMembers.html

Reply:
corrected